ui/plugin-system/src/components/MultiQueryEditor/MultiQueryEditor.tsx

```diff
@@ -15,36 +15,55 @@ import { useState } from 'react';
 import { produce } from 'immer';
 import { Button, Stack } from '@mui/material';
 import AddIcon from 'mdi-material-ui/Plus';
-import { TimeSeriesQueryDefinition, QueryDefinition } from '@perses-dev/core';
-import { TIME_SERIES_QUERY_KEY, useListPluginMetadata, usePlugin, usePluginRegistry } from '../../runtime';
-import { TimeSeriesQueryInput } from './TimeSeriesQueryInput';
+import { TimeSeriesQueryDefinition, QueryDefinition, QueryPluginType } from '@perses-dev/core';
+import { useListPluginMetadata, usePlugin, usePluginRegistry } from '../../runtime';
+import { QueryEditorContainer } from './QueryEditorContainer';
 
-export interface TimeSeriesQueryEditorProps {
-  queries?: TimeSeriesQueryDefinition[];
+export interface MultiQueryEditorProps {
+  queryTypes: QueryPluginType[];
+  queries?: QueryDefinition[];
   onChange: (queries: QueryDefinition[]) => void;
 }
 
-export function TimeSeriesQueryEditor({ queries = [], onChange }: TimeSeriesQueryEditorProps) {
+function useDefaultQueryDefinition(queryTypes: QueryPluginType[]): QueryDefinition {
   // Build the default query plugin
+  // This will be used only if the queries are empty, to open a starting query
+
+  // Firs the default query type
+  const defaultQueryType = queryTypes[0]!;
+
+  // Then the default plugin kind
   // Use as default the plugin kind explicitly set as default or the first in the list
+  const { data: queryPlugins } = useListPluginMetadata(queryTypes);
   const { defaultPluginKinds } = usePluginRegistry();
-  const { data: timeSeriesPlugins } = useListPluginMetadata(TIME_SERIES_QUERY_KEY);
-  const defaultTimeSeriesQueryKind = defaultPluginKinds?.[TIME_SERIES_QUERY_KEY] ?? timeSeriesPlugins?.[0]?.kind ?? '';
-  const { data: defaultQueryPlugin } = usePlugin(TIME_SERIES_QUERY_KEY, defaultTimeSeriesQueryKind, {
+  const defaultQueryKind = defaultPluginKinds?.[defaultQueryType] ?? queryPlugins?.[0]?.kind ?? '';
```

Comment:
Still works for Markdown panel too?

Reply:
MultiQueryEditor is not used in markdown panel as in this panel, there is a hideQueryEditor boolean. 
Oh! now I'm realizing that we might be able to remove this hideQueryEditor parameter to replace it by queryTypes is empty condition

Reply:
It makes me think as well that we might be able to remove the supportedQueryTypes field in favor of a regular tab 

from this 
```ts
export const TimeSeriesChart: PanelPlugin<TimeSeriesChartOptions> = {
  PanelComponent: TimeSeriesChartPanel,
  panelOptionsEditorComponents: [{ label: 'Settings', content: TimeSeriesChartOptionsEditorSettings }],
  supportedQueryTypes: ['TimeSeriesQuery'],
  createInitialOptions: createInitialTimeSeriesChartOptions,
};
```

To this (not sure it compiles but you see the idea)
```ts
function buildQueryEditor(queryTypes) {
  return function (props) {
    return <MultiQueryEditor queryTypes={queryTypes} {...props} />
  }

}
/**
 * The core TimeSeriesChart panel plugin for Perses.
 */
export const TimeSeriesChart: PanelPlugin<TimeSeriesChartOptions> = {
  PanelComponent: TimeSeriesChartPanel,
  panelOptionsEditorComponents: [
    { label: 'Query', content: buildQueryEditor(["TimeSeriesQuery"]) },
    { label: 'Settings', content: TimeSeriesChartOptionsEditorSettings }
  ],
  createInitialOptions: createInitialTimeSeriesChartOptions,
};
```

Reply:
Yeah that could be a lot to expose to plugin developers, although I guess we do already expose a lot as is

Now that I’m thinking about it, I wonder if putting supported data types on the actual metadata here where the plugin is registered would have made this any easier: https://github.com/perses/perses/blob/main/ui/panels-plugin/plugin.json#L10

Reply:
Yes actually this would even be complicate for me to manage the state from the spec editor if I'd do that. 

Regarding the place of supportedQueryTyprs we thought about it with Antoine and having it in the plugin.json might allow the backend to make some validation (if it exposes it, it can read it) that wouldn't be a bad idea